### DISCUSSION TITLE
Help with rendering multiple Geodes into an Image

### DISCUSSION THREAD

**Discussion by kohlisch-git** [openscenegraph/OpenSceneGraph#1257] [Q&A] [answered]
Hi everyone,
I am new to OpenSceneGraph and OpenGL and need Help with rendering multiple Objects into an image.
I have set up my camera with pre render with a frame buffer object and attached the color buffer to the image, and set the clear mask for color and depth buffer.  
If the scene only has one child, everything works just fine, however if add another child to the scene, the image turns black.

My scenegraph is set up as root-----camera------scene------child1
. . . . . . . . . . . . . . . . . . . . . . . . . . . |------scene . . . . . . . .|-------child2 

The root is set as scene data for the viewer, the duplicate of scene is, so that i can observe what is happening in 3d.

I have no idea, why the image turns black and what I am doing wrong. 
Is the second object rendered on top of the first object? Is the depth buffer considered for each colored pixel and only the closest thing is rendered?

I would really appreciate some help or hints!
Thanks, 
Niklas

Update:
Both children are drawn, however both are drawn completely in black. Child1 is originally a textured object, child2 is actually black.

I should note here, that the objects are dynamic and their vertexes and texture is changing between frames.  I do this by modifying the object and setting basically everything to dirty().

**Comment by robertosfield** [comment]
It's 2023 and folks still learning OSG/OpenGL...  might I suggest that a more modern scene graph and low level API might be a little more future proof.

As for objects turning black as the application, data and your combination of OS/machine are all only on your machine we can't provide anything other than general suggestions.  Try the same scene graph rendering to a on screen window to see what the results are,  Also check the console for OpenGL warnings.

**Comment by kohlisch-git** [comment]
Hello Robert and thanks for your answer.

I could identify the problem, but not yet solve it. Since child1 is textured, I did not define color or color binding (I have added it later, but the color is still set to black). The second child has color and color binding enabled and is drawn in black.
When rendering child1 the color seems to be taken from child2 and it is seemingly drawn on top  of the texture. How could I stop this from happening?

These are My StateSets for child1:
`int values = osg::StateAttribute::OFF | osg::StateAttribute::PROTECTED;
osg_mesh->getOrCreateStateSet()->setAttribute(new osg::PolygonMode(  osg::PolygonMode::FRONT_AND_BACK, osg::PolygonMode::FILL), values );
osg_mesh->getOrCreateStateSet()->setMode( GL_LIGHTING, values );
osg_mesh->getOrCreateStateSet()->setTextureAttributeAndModes(0, osg_tex,  osg::StateAttribute::ON | osg::StateAttribute::OVERRIDE);`

**Reply by robertosfield** [reply]
I'm afraid I can't provide fine grained support for the OSG, I haven't actively worked on it for 5 years, the project is in maintenance mode for existing users.  New users should be using newer and supported APIs.

**Reply by kohlisch-git** [reply, marked as answer]
Thanks for your answer anyways.

I was finally able to resolve this issue using material instead of color.
